api/agent.go

```diff
@@ -55,17 +55,15 @@ func (a *Agent) Self() (map[string]map[string]interface{}, error) {
 // populateCache is used to insert various pieces of static
 // data into the agent handle. This is used during subsequent
 // lookups for the same data later on to save the round trip.
-func (a *Agent) populateCache(info map[string]map[string]interface{}) {
+func (a *Agent) populateCache(self *AgentSelf) {
 	if a.nodeName == "" {
-		a.nodeName, _ = info["member"]["Name"].(string)
+		a.nodeName = self.Member.Name
 	}
-	if tags, ok := info["member"]["Tags"].(map[string]interface{}); ok {
-		if a.datacenter == "" {
-			a.datacenter, _ = tags["dc"].(string)
-		}
-		if a.region == "" {
-			a.region, _ = tags["region"].(string)
-		}
+	if a.datacenter == "" {
+		a.datacenter, _ = self.Member.Tags["dc"]
```

Comment:
[nitpick] Not sure why we used the two-form assignment here when we just ignore the second `ok` value... Feel free to remove the `, _` (or not; no big deal)